### PR TITLE
Implement the RenameTown game command

### PR DESCRIPTION
**Pull request by AaronVanGeffen**

### CODE REVIEW

File: src/OpenLoco/GameCommands/RenameTown.cpp [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+        static char* renameStringBuffer = reinterpret_cast<char*>(renameBuffer);
+
+        // Ensure the new name isn't empty.
+        if (strlen(renameStringBuffer) == 0)
+            return 0;
+
+        // Figure out the current name for this town.
+        char currentTownName[256] = "";
+        auto town = TownManager::get(_townId);
+        StringManager::formatString(currentTownName, town->name);
```
Comment (AaronVanGeffen) [inline src/OpenLoco/GameCommands/RenameTown.cpp:67]: I'm a bit puzzled why this doesn't always work. On one save it works fine, on another it yields:
```
terminate called after throwing an instance of 'std::out_of_range'
  what():  formatString: invalid string id: 52428
```

File: src/OpenLoco/GameCommands/RenameTown.cpp [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+            renameBuffer[arrayIndex * 3] = buffer0;
+            renameBuffer[arrayIndex * 3 + 1] = buffer1;
+            renameBuffer[arrayIndex * 3 + 2] = buffer2;
+        }
+
+        // Applying the buffer?
+        if (index != 0)
+            return 0;
+
+        static char* renameStringBuffer = reinterpret_cast<char*>(renameBuffer);
```
Comment (duncanspumpkin) [inline src/OpenLoco/GameCommands/RenameTown.cpp:58]: Why didn't you stick to the style of rename vehicle. Create another empty buffer for this and memcpy into it.
```
      char renameStringBuffer[37] = "";
        memcpy(renameStringBuffer, staticRenameBuffer, sizeof(staticRenameBuffer));
        renameStringBuffer[36] = '\0';```

File: src/OpenLoco/GameCommands/RenameTown.cpp [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+        // Ensure the new name isn't empty.
+        if (strlen(renameStringBuffer) == 0)
```
Comment (duncanspumpkin) [inline src/OpenLoco/GameCommands/RenameTown.cpp:61]: as it stands this is unsafe the buffer is not null terminated.

File: src/OpenLoco/GameCommands/RenameTown.cpp (hunk 1 above)
Comment (duncanspumpkin) [reply src/OpenLoco/GameCommands/RenameTown.cpp:67]: because you aren't passing in the arguments.

Comment (AaronVanGeffen) [reply src/OpenLoco/GameCommands/RenameTown.cpp:67]: The original is passing `common_format_args`, but confusingly, it doesn't appear to be setting anything to them at all. That's why I left them out. What am I missing?

File: src/OpenLoco/GameCommands/RenameTown.cpp (hunk 2 above)
Comment (AaronVanGeffen) [reply src/OpenLoco/GameCommands/RenameTown.cpp:58]: My reasoning was that we null-terminate the string in the text input window already, and I didn't like having to copy unnecessarily. However, you're right that that can't be relied on while networked, of course. In that case, the solution proposed isn't great either, though: you could end up leaving 36 chars of (format code) garbage before the null byte. Thoughts on that?

File: src/OpenLoco/GameCommands/RenameTown.cpp (hunk 3 above)
Comment (AaronVanGeffen) [reply src/OpenLoco/GameCommands/RenameTown.cpp:61]: See above; my reasoning was that we null-terminate the string in the text input window already.

File: src/OpenLoco/GameCommands/RenameTown.cpp (hunk 1 above)
Comment (AaronVanGeffen) [reply src/OpenLoco/GameCommands/RenameTown.cpp]: I have added them now, but the error persists.

File: src/OpenLoco/GameCommands/RenameTown.cpp (hunk 2 above)
Comment (duncanspumpkin) [reply src/OpenLoco/GameCommands/RenameTown.cpp:58]: That's not going to work when you input a 36 character string.

File: src/OpenLoco/GameCommands/RenameTown.cpp (hunk 1 above)
Comment (AaronVanGeffen) [reply src/OpenLoco/GameCommands/RenameTown.cpp]: I just realised the error occurs when clicking to rename (i.e. text input window opens), not when performing the rename. So this is an existing bug.

File: src/OpenLoco/GameCommands/RenameTown.cpp [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+        renameStringBuffer[36] = '\0';
+
+        // Ensure the new name isn't empty.
+        if (strlen(renameStringBuffer) == 0)
+            return 0;
+
+        // Figure out the current name for this town.
+        char currentTownName[256] = "";
+        auto town = TownManager::get(_townId);
+        auto args = FormatArguments::common();
```
Comment (duncanspumpkin) [inline src/OpenLoco/GameCommands/RenameTown.cpp:69]: you were quite right there is no need to pass common_format args in here

**Review by duncanspumpkin — APPROVED** [review]

**Pull request merged**